### PR TITLE
SYSLIB reference

### LINKED ISSUE #21157: Add SYSLIB warning reference docs
These docs will be linked from the build output when the warning is generated.

### PR DESCRIPTION
**Pull request by gewarren**
[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/syslib-obsoletions?branch=pr-en-us-21156).

Fixes #21157.

### CODE REVIEW

File: docs/core/compatibility/syslib0008.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: SYSLIB0008 warning
+description: Learn about the obsoletion that generates compile-time warning SYSLIB0008.
+ms.topic: reference
+ms.date: 10/20/2020
+---
+# SYSLIB0008: CreatePdbGenerator is not supported
+
+The <xref:System.Runtime.CompilerServices.DebugInfoGenerator.CreatePdbGenerator?displayProperty=nameWithType> API is marked obsolete, starting in .NET 5.0. Using this API generates warning `SYSLIB0008` at compile time.
```
Comment (gewarren) [inline docs/core/compatibility/syslib0008.md:9]: Anything else we can add to this article?

File: docs/core/compatibility/syslib0012.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+title: SYSLIB0012 warning
+description: Learn about the obsoletions that generate compile-time warning SYSLIB0012.
+ms.topic: reference
+ms.date: 10/20/2020
+---
+# SYSLIB0012: Assembly.CodeBase and Assembly.EscapedCodeBase are obsolete
+
+The following APIs are marked as obsolete, starting in .NET 5.0. Using them in code generates warning `SYSLIB0012` at compile time.
```
Comment (gewarren) [inline docs/core/compatibility/syslib0012.md:9]: Anything else we can add to this article, e.g. why they are obsolete?

File: docs/core/compatibility/syslib0001.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+title: SYSLIB0001 warning
+description: Learn about the obsoletions that generate compile-time warning SYSLIB0001.
+ms.topic: reference
+ms.date: 10/20/2020
+---
+# SYSLIB0001: The UTF-7 encoding is insecure
```
Comment (BillWagner) [inline docs/core/compatibility/syslib0001.md:7]: I like the way you've include the warning code, the warning text and the key phrases in the title, H1, and descriptions. That's really well done. (Applies to all these files)

File: docs/core/compatibility/syslib0003.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+---
+# SYSLIB0003: Code access security is not supported
+
+[Code access security (CAS)](../../framework/data/adonet/code-access-security.md) is an unsupported, legacy technology. The infrastructure to enable CAS, which exists only in .NET Framework 2.x - 4.x, is deprecated and not receiving servicing or security fixes.
+
+As a result, most code access security (CAS)-related types in .NET are obsolete, starting in .NET 5.0. This includes CAS attributes, such as <xref:System.Security.Permissions.SecurityPermissionAttribute>, CAS permission objects, such as <xref:System.Net.SocketPermission>, <xref:System.Security.Policy.EvidenceBase>-derived types, and other supporting APIs. Using these APIs generates warning `SYSLIB0003` at compile time.
+
+The complete list of obsolete CAS APIs is as follows:
+
+- <xref:System.AppDomain.PermissionSet?displayProperty=fullName>
```
Comment (BillWagner) [inline docs/core/compatibility/syslib0003.md:15]: With all these APIs, should we append the moniker for framework 4.x? Or use the default moniker?

File: docs/core/compatibility/syslib0006.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+title: SYSLIB0006 warning
+description: Learn about the obsoletions that generate compile-time warning SYSLIB0006.
+ms.topic: reference
+ms.date: 10/20/2020
+---
+# SYSLIB0006: Thread.Abort is not supported
```
Comment (BillWagner) [inline docs/core/compatibility/syslib0006.md:7]: Hmmm. I did a quick check in our conceptual docs and we have ~10 articles that mention `Thread.Abort`. Most seem to discourage its use. Should we link those articles back to this one?

File: docs/core/compatibility/syslib0008.md (hunk 1 above)
Comment (BillWagner) [reply docs/core/compatibility/syslib0008.md:9]: ping @jaredpar @tmat 

One of them might know more about this API.

File: docs/core/compatibility/syslib0010.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# SYSLIB0010: Unsupported remoting APIs
+
+[.NET remoting](/previous-versions/dotnet/netframework-1.1/kwdt6w2k(v=vs.71)) is a legacy technology, and the infrastructure exists only in .NET Framework. The following remoting-related APIs are marked as obsolete, starting in .NET 5.0. Using them in code generates warning `SYSLIB0010` at compile time.
+
+- <xref:System.MarshalByRefObject.GetLifetimeService?displayProperty=nameWithType>
+- <xref:System.MarshalByRefObject.InitializeLifetimeService?displayProperty=nameWithType>
+
+## Workaround
+
+Consider using WCF or HTTP-based REST services to communicate with objects in other applications or across machines. For more information, see [.NET Framework technologies unavailable on .NET Core](../porting/net-framework-tech-unavailable.md).
```
Comment (BillWagner) [inline docs/core/compatibility/syslib0010.md:16]: Should we mention gRPC here?

**Review by BillWagner — COMMENTED** [review]
This looks great @gewarren 

I pinged a couple folks on your one question, and had a couple suggestions.

(Commenting rather than approving pending decision on those discussions)

File: docs/core/compatibility/syslib0001.md (hunk 3 above)
Comment (gewarren) [reply docs/core/compatibility/syslib0001.md:7]: Thanks Bill. Actually I changed the headings a bit, mostly to shorten them. Do you think they should match the warning text exactly? Those are listed [here](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Obsoletions.cs).

**Review by jeffhandley — APPROVED** [review]
These all look good to me (thanks for the Preview link). I appreciate the questions/comments already made, but I have nothing to add.

@GrabYourPitchforks -- can you review too?

**Comment by gewarren** [comment]
I also see SYSLIB0013 and SYSLIB0014 in the source code. Are those obsoletions for RTM?

File: docs/core/compatibility/syslib0002.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+---
+title: SYSLIB0002 warning
```
Comment (gewarren) [inline docs/core/compatibility/syslib0002.md:2]:
```suggestion
title: SYSLIB0002 error
```

File: docs/core/compatibility/syslib0002.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+description: Learn about the obsoletion that generates compile-time warning SYSLIB0002.
```
Comment (gewarren) [inline docs/core/compatibility/syslib0002.md:3]:
```suggestion
description: Learn about the obsoletion that generates compile-time error SYSLIB0002.
```

**Comment by jeffhandley** [comment]
> I also see SYSLIB0013 and SYSLIB0014 in the source code. Are those obsoletions for RTM?

No, those did not make 5.0; they'll be in 6.0.

File: docs/core/compatibility/syslib0003.md (hunk 4 above)
Comment (gewarren) [reply docs/core/compatibility/syslib0003.md:15]: I think it's fine to use the default moniker. Those are the ones that will show the obsolete pink/red caution box.

File: docs/core/compatibility/syslib0006.md (hunk 5 above)
Comment (gewarren) [reply docs/core/compatibility/syslib0006.md:7]: I logged an issue for this.

**Pull request merged**